solidity/random-beacon/contracts/libraries/Relay.sol

```diff
@@ -133,23 +127,36 @@ library Relay {
             "Invalid entry"
         );
 
-        // If the soft timeout has been exceeded apply stake slashing for
-        // all group members. Note that `getSlashingFactor` returns the
-        // factor multiplied by 1e18 to avoid precision loss. In that case
-        // the final result needs to be divided by 1e18.
-        slashingAmount =
-            (getSlashingFactor(self, dkgGroupSize) *
-                self.relayEntrySubmissionFailureSlashingAmount) /
-            1e18;
-
         emit RelayEntrySubmitted(self.currentRequestID, msg.sender, entry);
 
         self.previousEntry = AltBn128.g1Unmarshal(entry);
         self.currentRequestID = 0;
         self.currentRequestGroupID = 0;
         self.currentRequestStartBlock = 0;
+    }
+
+    /// @notice Calculates the slashing amount for all group members.
+    /// @dev Must be used when a soft timeout was hit.
+    /// @return Amount by which group members should be slashed
+    ///         in case the relay entry was submitted after the soft timeout.
+    function calculateSlashingAmount(Data storage self)
+        internal
+        returns (uint256)
+    {
+        uint256 softTimeoutBlock = self.currentRequestStartBlock +
+            (dkgGroupSize * self.relayEntrySubmissionEligibilityDelay);
```

Comment:
`getSlashingFactor` was accepting `_groupSize` before. Don't we need the group size to be configurable in `calculateSlashingAmount`?

Reply:
`_groupSize` was there for tests only and I believe it was added when all the tests were executed very slowly, before the "snapshots". It is not needed for production code. Now in total, we have 3 tests that validate the calculation of the slashing amount which also use snapshots to make them faster. Having a group number of 64 members actually makes a test more realistic and also I didn't notice that much slow execution. I think we can avoid changing the production code just for 3-4 tests.